packages/components/src/lib/controllers/settings-controller.js

```diff
@@ -0,0 +1,264 @@
+    this._store.registerModule('settings', {
+      // All stores of modules are namespaced
+      namespaced: true,
+      state: {
+        // these counters are used to enable the settings ui components
+        // to redraw themselves when settings are reset or reloaded
+        // it might be better if all settings were made into
+        // state variables but for now state is monitored at the domain level
+        uiResetCounter: 0,
+        featureResetCounter: 0,
+        resourceResetCounter: 0
+      },
+      mutations: {
+        incrementUiResetCounter (state) {
+          state.uiResetCounter += 1
+        },
+
+        incrementFeatureResetCounter (state) {
+          state.featureResetCounter += 1
+        },
+
+        incrementResourceResetCounter (state) {
+          state.resourceResetCounter += 1
+        }
+      }
+    })
+  }
+
+  /**
+   * Loads an application wide configuration file in a JSON format.
+   *
+   * @returns {Promise<object>} - A promise that is resolved with an app config response.
+   */
+  async requestAppOptions () {
+    const configUrl = `${this._configServiceUrl}?clientId=${encodeURIComponent(this._clientId)}&appName=${encodeURIComponent(this._appName)}` +
+      `&appVersion=${encodeURIComponent(this._appVersion)}&buildBranch=${encodeURIComponent(this._branch)}` +
+      `&buildNumber=${encodeURIComponent(this._buildNumber)}`
+    const request = new Request(configUrl)
+    return fetch(request)
+  }
+
+  /**
+   * initialize the options using the supplied storage adapter class
+   *
+   * @param {Function<StorageAdapter>} StorageAdapter the adapter class to instantiate
+   * @param {object} authData optional authentication data if the adapter is one that requires it
+   * @returns Promise[] an array of promises to load the options data from the adapter
+   */
+  initOptions (StorageAdapter = this._storageAdapter, authData = null) {
+    this._featureOptions = new Options(this._featureOptionsDefaults, new StorageAdapter(this._featureOptionsDefaults.domain, authData))
+    this._resourceOptions = new Options(this._resourceOptionsDefaults, new StorageAdapter(this._resourceOptionsDefaults.domain, authData))
+    this._uiOptions = new Options(this._uiOptionsDefaults, new StorageAdapter(this._uiOptionsDefaults.domain, authData))
+    return [this._featureOptions.load(), this._resourceOptions.load(), this._uiOptions.load()]
+  }
+
+  initNonConfigurableOptions () {
+    this._lookupResourceOptions = new Options(this._resourceOptionsDefaults, new this._storageAdapter(this._resourceOptionsDefaults.domain))
+    this._siteOptions = [] // eslint-disable-line prefer-const
+    for (const site of this._siteOptionsDefaults) {
+      for (const domain of site.options) {
+        const siteOpts = new Options(domain, new this._storageAdapter(domain.domain)) // eslint-disable-line new-cap
+        this._siteOptions.push({ uriMatch: site.uriMatch, resourceOptions: siteOpts })
+      }
+    }
+  }
```

Comment:
let's get rid of it

Reply:
OK

Reply:
well, see previous comment -- I guess we are keeping siteOptions